### DISCUSSION TITLE
Confused about Models and factories

### DISCUSSION THREAD

**Discussion by vnugent** [adonisjs/core#1744] [Help] [answered]
I have followed the docs and was able to create a controller, a simple test response and db migration.  What I am missing is how to tie all of the pieces together with fake data.  My rest api template has files in App/Models.  I can't find it documented anywhere.  Do I need it?

Here's my controller
```FooController.js
  async show({ params, request, response, view }) {
    response.json({ content: "cool" + params.id });
   
    // what steps do I need to generate fake data and return them here?
    // migration code already creates Foos table

  }


```

**Comment by thetutlage** [comment, marked as answer]
Okay. Let's do it step by step.

### Models
Make sure you have models created inside `app/Models` directory. The job of a model is to query a single database table. For example: The `User` will conventionally query the `users` database table. For Example:

```ts
await User.all() // select * from users;
await User.findBy(params.id) // select * from users where id = ?;
```

### Fake data
Fake data can come from anywhere. For example: You can insert fake data from a JSON file to your database. Or you can use factories to do that.

Now of course adding fake data to the database action has to happen without creating any route (HTTP endpoint). So we will use seeders for that. Here's the documentation on seeders https://preview.adonisjs.com/guides/database/seeds

Things to keep in mind

- A seeder is a class that you can execute by running `node ace db:seed` command.
- You can write any code inside this class run method. You are not forced to use factories here. In fact, you can import the model and insert values by hardcoding them
- If you need randomisation in your fake data, then you can look into creating factories.

**Comment by vnugent** [comment]
Thanks.  Eventually I found out about Model in the Lucid section further down. I think what I (or many newbies) need is a task-oriented Getting Started guide without being exposed to too many advanced features and options.  Right what I need to know to build an end-to-end API endpoint connecting to a Postgres db, the Adonis way, is sprinkled throughout multiple sections.  Just some constructive feedback.  When I get a better handle of the framework I'd be happy to contribute a simple tutorial.

**Reply by RomainLanz** [reply]
Have you watched [our Tour Video ](https://www.youtube.com/watch?v=TysfaNcFX_Y)that present the framework?

**Reply by vnugent** [reply]
> Have you watched [our Tour Video ](https://www.youtube.com/watch?v=TysfaNcFX_Y)that present the framework?

I have not... as a dev I'd prefer to read how-tos and cut and paste working examples over watching video. But I'll give it a try.  At the moment I'm looking at Lucid section... I'm not sure how or should I define columns for my model.
